site/src/App/guides/testing-guide.tsx

```diff
@@ -0,0 +1,251 @@
+        experience this causes people to grow accustomed to them changing and
+        ignore them all together. This is especially relevant when using Braid,
+        as the components will change frequently, causing many snapshots to
+        break through your app, even though it remains functional.
+      </Text>
+
+      <Text>
+        Referencing our definition of a &quot;good test&quot;, snapshots
+        don&apos;t meet any of this criteria. Instead of documenting what a
+        component should do. They document what a component does right now.
+        Instead of being resilient to refactors, they break even when
+        meaningless changes occur (e.g. class names change).
+      </Text>
+
+      <Text>
+        What&apos;s often raised as a solution to some of the above is shallow
+        rendered snapshots. Check out &quot;
+        <ExternalLink href="https://kentcdodds.com/blog/why-i-never-use-shallow-rendering">
+          Why I Never Use Shallow Rendering
+        </ExternalLink>
+        &quot; for great article on the issues with shallow rendering.
+      </Text>
+
+      <Text>
+        Note: The above is specifically discussing react component snapshots.
+        Snapshot testing is a great tool when applied to the appropriate
+        use-case.
+      </Text>
+
+      <Heading level="3">What about integration tests?</Heading>
+
+      <Text>
+        We suggest all teams have at least a small number of integration tests
+        making sure their core flows work as expected. Here&apos;s a couple of
+        great libraries to consider:
+      </Text>
+
+      <BulletList>
+        <Bullet>
+          <ExternalLink href="https://www.cypress.io/">Cypress</ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://testing-library.com/docs/cypress-testing-library/intro">
+            Cypress Testing Library
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://github.com/puppeteer/puppeteer">
+            Puppeteer
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://github.com/smooth-code/jest-puppeteer">
+            jest-puppeteer
+          </ExternalLink>
+        </Bullet>
+        <Bullet>
+          <ExternalLink href="https://testing-library.com/docs/pptr-testing-library/intro">
+            Puppeteer Testing Library
+          </ExternalLink>
+        </Bullet>
+      </BulletList>
+
+      <Heading level="3">
+        How do I query for elements rendered by Braid?
+      </Heading>
+
+      <Text>
+        If you need to query for an element rendered via a Braid component we
+        suggest using{' '}
+        <ExternalLink href="https://testing-library.com/docs/dom-testing-library/api-queries#byrole">
+          role based querying
+        </ExternalLink>
+        . Targetting labels and ids is also a good option. Some Braid components
+        also allow passing{' '}
```

Comment:
drop "also"

Reply:
Why do only "some" Braid components support that testing attribute? If it's something we recommend shouldn't be more widely supported?